bot.py

```diff
@@ -47,14 +51,29 @@ async def on_message(msg):
         except DraftDataParseError as ex:
             await msg.channel.send(ex.message)
 
+
 async def parse_submission(msg):
+    # Regex for Win-Loss YYYY-MM-DD HH:MM
+    msg_re = re.compile(
+        r'(?:\b(?P<wins>[0-3])-[0-3]\b)?(?:\s)?'
+        r'(?:\b(?P<year>[0-9]{4})-(?P<month>1[0-2]|0[0-9])-(?P<day>0[1-9]|[1-2][0-9]|3[0-1])\b)?'
+        r'(?:\s)?(?:\b(?P<hour>[0-1][0-9]|2[0-4]):(?P<minute>[0-5][0-9])\b)?'
+        )
+    # Assign the parameters found.
+    msg_params = msg_re.search(msg.content).groupdict(default="")
+    wins = msg_params.pop("wins")
+    if wins != "":
+        wins = int(wins)
+    date = msg.created_at
+    date = date.replace(**{key: int(val) for key, val in msg_params if val != ""})
+    # Handle attached file.
```

Comment:
Tried to make this clearer with one regex split into rows by record, date and time.